### PR TITLE
Irmaku/update node18

### PR DESCRIPTION
**Pull request by irma-kurnia-alaska**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Resolves: #166 , #167 , #168 , #169 , #171

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

- [x] Revision of an existing capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    attributes:
+      value: >
+        [![See it on
+        NPM!](https://img.shields.io/npm/v/@alaskaairux/auro-hyperlink?style=for-the-hyperlink&color=orange)](https://www.npmjs.com/package/@alaskaairux/auro-hyperlink)
+  - type: textarea
+    id: details
+    attributes:
+      label: Please describe the bug
+      description: List out the steps to reproduce the behavior
+      placeholder: 1. Go to '...' 2. Click on '....' 3. Scroll down to '....' 4. See error
```
Comment (blackfalcon) [inline .github/ISSUE_TEMPLATE/bug_report.yml:27]:
```suggestion
      description: List out the steps to reproduce the behavior and include screenshots
      placeholder: |
        1. Go to '...'
        2. Click on '....'
        3. Scroll down to '....'
        4. See error
```

File: .npmignore [hunk 2]
```diff
@@ -18,6 +18,8 @@ demo/**/*.scss
 demo/**/*.html
 demo/**/*.md
 
+dist/es5.js
```
Comment (blackfalcon) [inline .npmignore:21]: This is not correct. Please update file to look like 

https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/.npmignore.temp

File: demo/index.html [hunk 3]
```diff
@@ -20,18 +20,18 @@
...
       type="text/css"
-      href="https://unpkg.com/prismjs@1.24.1/themes/prism.css"
+      href="https://cdn.jsdelivr.net/npm/prismjs@1.24.1/themes/prism.css"
     />
-    <link rel="stylesheet" href="https://unpkg.com/@alaskaairux/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
-    <link rel="stylesheet" type="text/css" href="https://unpkg.com/@alaskaairux/webcorestylesheets@latest/dist/demoWrapper.css" />
-    <link rel="stylesheet" type="text/css" href="https://unpkg.com/@alaskaairux/webcorestylesheets@latest/dist/elementDemoStyles.css" />
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@alaskaairux/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
+    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/@alaskaairux/webcorestylesheets@latest/dist/demoWrapper.css" />
+    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/@alaskaairux/webcorestylesheets@latest/dist/elementDemoStyles.css" />
```
Comment (blackfalcon) [inline demo/index.html:27]:
```suggestion
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@alaskaairux/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/@aurodesignsystem/webcorestylesheets@latest/dist/demoWrapper.css" />
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/@aurodesignsystem/webcorestylesheets@latest/dist/elementDemoStyles.css" />
```

File: package.json [hunk 4]
```diff
@@ -30,62 +30,59 @@
...
+    "postcss-discard-comments": "^6.0.0",
     "postcss-selector-replace": "^1.0.2",
     "prismjs": "^1.29.0",
-    "rollup": "^3.20.0",
+    "rollup": "^3.21.5",
     "rollup-plugin-serve": "^2.0.2",
-    "sass": "^1.59.3",
-    "semantic-release": "^20.1.3",
-    "stylelint": "^15.3.0",
+    "sass": "^1.62.1",
```
Comment (blackfalcon) [inline package.json:69]: Please add the following for properly using Sass

https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/package.json#L75-L79

File: package.json [hunk 5]
```diff
@@ -136,28 +133,33 @@
     "web components"
   ],
   "scripts": {
-    "build": "npm-run-all build:sass sass:render build:api test build:markdownDocs bundler postinstall",
-    "build:api": "wca analyze 'src/auro-hyperlink.js' --outFiles docs/api.md",
+    "build": "npm-run-all build:sass sass:render cssLint dist:js bundler postinstall build:api types",
+    "build:api": "wca analyze 'src/auro-badge.js' --outFiles docs/api.md",
```
Comment (blackfalcon) [inline package.json:137]:
```suggestion
    "build:api": "wca analyze 'src/auro-hyperlink.js' --outFiles docs/api.md",
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Comment by irma-kurnia-alaska** [comment]
Hi @blackfalcon,

I will updated as your feedback. 
Thank you.

File: package.json [hunk 6]
```diff
@@ -30,62 +30,64 @@
...
+    "@alaskaairux/icons": "^4.27.0",
     "@alaskaairux/webcorestylesheets": "^3.7.3",
     "@aurodesignsystem/eslint-config": "^1.3.0",
-    "@babel/eslint-parser": "^7.21.3",
-    "@commitlint/cli": "^17.4.4",
-    "@commitlint/config-conventional": "^17.4.4",
-    "@open-wc/testing": "^3.1.7",
-    "@rollup/plugin-node-resolve": "^15.0.1",
-    "@semantic-release/changelog": "^6.0.2",
+    "@babel/eslint-parser": "^7.21.8",
```
Comment (blackfalcon) [inline package.json:37]:
```suggestion

```

Please cross-reference between this package.json and the one in the Generator. 

https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/package.json

File: package.json [hunk 7]
```diff
@@ -19,8 +19,8 @@
     "node": ">=18.15.0"
   },
   "dependencies": {
-    "chalk": "^4.1.2",
-    "lit": "^2.6.1"
+    "chalk": "^5.2.0",
```
Comment (blackfalcon) [inline package.json:22]: Updating this to v5+, there are additional things that need to be updated as well. Please see the following files from the generator. 

1. https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/scripts/generateDocs.js
2. https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/scripts/pre-commit.mjs

File: package.json [hunk 8]
```diff
@@ -136,28 +138,33 @@
     "web components"
   ],
   "scripts": {
-    "build": "npm-run-all build:sass sass:render build:api test build:markdownDocs bundler postinstall",
+    "build": "npm-run-all build:sass sass:render cssLint dist:js bundler postinstall build:api types",
     "build:api": "wca analyze 'src/auro-hyperlink.js' --outFiles docs/api.md",
     "build:demo": "node ./scripts/prepForBuild",
-    "build:dev:assets": "npm-run-all build:sass:component postCss:component sass:render build:markdownDocs",
+    "build:dev:assets": "npm-run-all build:sass:component postCss:component sass:render",
     "build:markdownDocs": "node scripts/generateDocs.js",
```
Comment (blackfalcon) [inline package.json:145]:
```suggestion
    "build:docs": "node scripts/generateDocs.js",
```

File: scripts/removeNonRemPlugin.js [hunk 9]
```diff
@@ -1,27 +0,0 @@
-const postcss = require('postcss');
```
Comment (blackfalcon) [inline scripts/removeNonRemPlugin.js:1]: Also remove this file 

https://github.com/AlaskaAirlines/auro-hyperlink/blob/main/scripts/prepForBuild.js

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Review by blackfalcon — CHANGES_REQUESTED** [review]
I can't understand what happened with this commit

- https://github.com/AlaskaAirlines/auro-hyperlink/pull/170/commits/35f74bdecb34a6934122689757c37e397d50d1d3

There are a lot of things here that were not requested to have happen. Please explain.

File: demo/demo.md [hunk 10]
```diff
@@ -1,449 +1,86 @@
...
-
-#### Relative URLs
-
-Absolute URLs are not always preferred. When using relative URLs, use the `relative` attribute and `auro-hyperlink` will ignore the domain and the protocol.
-
-<div class="exampleWrapper">
-  Example of a <auro-hyperlink relative href="#">relative URL</auro-hyperlink>.
-</div>
-
+<!--
```
[the remark this answers is not in the document]
Comment (irma-kurnia-alaska) [reply demo/demo.md:1]: Hi @blackfalcon,

for commit id https://github.com/AlaskaAirlines/auro-hyperlink/commit/35f74bdecb34a6934122689757c37e397d50d1d3 when I updated the last feedback some file need to be updated. so I follow the WC generator template. I will check again. Thank you.

File: .stylelintrc [hunk 11]
```diff
@@ -1,38 +1,53 @@
 {
-  "extends": ["stylelint-config-recommended", "stylelint-config-idiomatic-order", "stylelint-config-standard-scss"],
+  "extends": [
```
Comment (blackfalcon) [inline .stylelintrc:2]: Remove the following deprecated rules. 

```
Deprecation warnings:
 - The "at-rule-name-newline-after" rule is deprecated.
 - The "at-rule-semicolon-space-before" rule is deprecated.
 - The "block-closing-brace-space-after" rule is deprecated.
 - The "block-opening-brace-newline-before" rule is deprecated.
 - The "declaration-block-semicolon-newline-before" rule is deprecated.
 - The "media-query-list-comma-newline-before" rule is deprecated.
 - The "no-empty-first-line" rule is deprecated.
 - The "number-leading-zero" rule is deprecated.
 - The "selector-list-comma-newline-before" rule is deprecated.
 - The "selector-list-comma-space-after" rule is deprecated.
 - The "string-quotes" rule is deprecated.
 - The "value-list-comma-newline-before" rule is deprecated.
```

File: package.json [hunk 12]
```diff
@@ -136,28 +133,33 @@
     "web components"
   ],
   "scripts": {
-    "build": "npm-run-all build:sass sass:render build:api test build:markdownDocs bundler postinstall",
+    "build": "npm-run-all build:sass sass:render scssLint dist:js bundler postinstall build:api types build:docs",
     "build:api": "wca analyze 'src/auro-hyperlink.js' --outFiles docs/api.md",
+    "build:ci": "npm-run-all sweep build",
     "build:demo": "node ./scripts/prepForBuild",
```
Comment (blackfalcon) [inline package.json:139]: Remove script

File: scripts/generateDocs.js [hunk 13]
```diff
@@ -1,20 +1,20 @@
...
 const markdownMagic = require('markdown-magic');
 const fs = require('fs');
+const https = require('https');
 
-const config = {
-  matchWord: 'AURO-GENERATED-CONTENT'
-}
+const readmeTemplateUrl = 'https://raw.githubusercontent.com/AlaskaAirlines/WC-Generator/master/componentDocs/README.md';
+const dirDocTemplates = './docTemplates';
+const readmeFilePath = dirDocTemplates + '/README.md';
```
Comment (blackfalcon) [inline scripts/generateDocs.js:8]: Please revert this whole file. Many of the features in this script this repo does not support. 

https://github.com/AlaskaAirlines/auro-hyperlink/blob/main/scripts/generateDocs.js

File: demo/demo.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (blackfalcon) [reply demo/demo.md:1]: Please revert to this state

https://github.com/AlaskaAirlines/auro-hyperlink/blob/main/demo/demo.md

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Comment by irma-kurnia-alaska** [comment]
Hi @blackfalcon, 

I will update the PR as your feedback, please review this, thank you.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 14]
```diff
@@ -0,0 +1,60 @@
+name: Feature request
```
Comment (blackfalcon) [inline .github/ISSUE_TEMPLATE/bug_report.yml:1]: Wrong name

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Comment by irma-kurnia-alaska** [comment]
Ok @Dale ***@***.***>, i will update. Thank you

Dapatkan Outlook untuk Android<https://aka.ms/AAb9ysg>
________________________________

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 14 above)
Comment (irma-kurnia-alaska) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Hi @blackfalcon,

I changed the name for this. Thank you.

File: scripts/generateDocs.js [hunk 15]
```diff
@@ -1,30 +1,32 @@
 const path = require('path');
-const chalk = require('chalk');
 const markdownMagic = require('markdown-magic');
 const fs = require('fs');
 
 const config = {
   matchWord: 'AURO-GENERATED-CONTENT'
-}
+};
```
Comment (blackfalcon) [inline scripts/generateDocs.js:7]: This whole document is incorrect? Please use code from https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/scripts/generateDocs.js

Also, when code is updated note the error of missing resources that were deleted in a previous commit.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: scripts/generateDocs.js (hunk 15 above)
Comment (irma-kurnia-alaska) [reply scripts/generateDocs.js]: Hi @blackfalcon,
I will fix your feedback, thank you.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
ERROR: ./demo/apiExamples.md file is missing

**Comment by irma-kurnia-alaska** [comment]
Hi @blackfalcon, 

i will fix this for the issue ERROR: ./demo/apiExamples.md file is missing. Sorry for this. When I checked the apiExamples is hidden in my machine and i will fix this. thank you.

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**